src/components/Modal/Modal.stories.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { withInfo } from '@storybook/addon-info';
+import { defaultTemplate } from '../../../storybook/decorators/storyTemplates';
+
+import {
+  MockModalManager,
+  basicExampleSource,
+} from './__mocks__/mockModalManager';
+
+import {
+  MockAboutModalManager,
+  aboutExampleSource,
+} from './__mocks__/mockAboutModalManager';
+
+const stories = storiesOf('Modal Overlay', module);
+
+const description = (
+  <p>
+    This component is based on React Bootstrap Modal component. See{' '}
+    <a href="https://react-bootstrap.github.io/components.html#modals">
+      React Bootstrap Docs
+    </a>{' '}
+    for complete Modal component documentation.
+  </p>
+);
+
+stories.addDecorator(
+  defaultTemplate({
+    title: 'Modal Overlay',
+    documentationLink:
+      'http://www.patternfly.org/pattern-library/forms-and-controls/modal-overlay/',
+    description: description,
+  }),
+);
+
+stories.add(
+  'Basic example',
+  withInfo({
+    source: false,
+    propTablesExclude: [MockModalManager],
+    text: (
+      <div>
+        <h1>Story Source</h1>
+        <pre>{basicExampleSource}</pre>
+      </div>
+    ),
+  })(() => <MockModalManager />),
+);
+
+stories.add(
+  'About Modal',
+  withInfo({
+    source: false,
+    propTablesExclude: [MockAboutModalManager],
+    text: (
+      <div>
+        <h1>Story Source</h1>
+        <pre>{aboutExampleSource}</pre>
+      </div>
+    ),
+  })(() => <MockAboutModalManager />),
+);
```

Comment:
I don't like the fact you are overriding the text, we are losing tons of benefits.
Some times something like this worked for me:
```js
() => MockAboutModalManager()
```

Reply:
* same in wizards

Reply:
Nice 👍 ! I will try it. This would be a huge relief...

Reply:
Any update?

Reply:
Unfortunately, it works only with stateless components :/
I guess there is no trivial solution, would be nice if storybook had an option to manually trigger knobs updates.

Reply:
@sharvit I think this might be something for us to consider in the future... not sure if it helps the React docgen/withInfo content... but it would help make the state more visible.
https://github.com/ndelangen/storybook-state